.github/workflows/deploy-docs-and-extensions.yml

```diff
@@ -44,9 +65,8 @@ jobs:
           cp -r extensions-site/build/client/* combined-build/v1/extensions/
 
       - name: Deploy to /gh-pages
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

Comment:
i think this will fix the issue we were seeing with PRs from forked repos failing on checks, example:
- https://github.com/block/goose/pull/886
- https://github.com/block/goose/actions/runs/13027268451/job/36352401784

Reply:
Yes it would